flytekit/common/nodes.py

```diff
@@ -219,8 +272,49 @@ def promote_from_model(cls, model):
         :param flytekit.models.core.workflow.Node model:
         :rtype: SdkNode
         """
-        raise _user_exceptions.FlyteAssertion("An SDK node cannot be instantiated merely from a data model object "
-                                              "because it must be contextualized within a workflow.")
+        id = model.id
+        # This should never be called
+        if id == "start-node" or id == "end-node":
+            _logging.warning("Should not call promote from model on a start node or end node {}".format(model))
+            return None
+
+        sdk_task_node, sdk_workflow_node = None, None
+        if model.task_node is not None:
+            sdk_task_node = SdkTaskNode.promote_from_model(model.task_node)
+        elif model.workflow_node is not None:
+            sdk_workflow_node = SdkWorkflowNode.promote_from_model(model.workflow_node)
+        else:
+            raise Exception("bad Node model")
```

Comment:
let's use the system exceptions `flytekit.common.exceptions.system` just so we can build better visibility in as we go through a common set of exceptions.

Reply:
will do, this was just a placeholder.